src/dns-test.js

```diff
@@ -293,19 +341,43 @@ async function sendQueries(nameservers_ipv4) {
  */
 function sendTelemetry(payload) {
     payload.measurementID = measurementID;
-    browser.telemetry.submitPing(TELEMETRY_TYPE, payload, TELEMETRY_OPTIONS);
+    console.log(payload);
+    // browser.telemetry.submitPing(TELEMETRY_TYPE, payload, TELEMETRY_OPTIONS);
+}
+
+async function fetchTest() {
+    // Unsure if we want to wrap the fetch in a try/catch. This would enable 
+    // us to handle network errors, in case that matters to us. I don't think 
+    // it does, though. I think all we want is to stop the measurement from 
+    // running if we don't get the response we expect. Also, if the fetch fails, 
+    // it doesn't make sense to send anything to telemetry, because that 
+    // may fail, too.
+    let response = await fetch("https://dnssec-experiment-moz.net/");
+    let responseOk = await response.ok;
+    let responseText = await response.text();
+    if (!(responseOk && responseText && responseText === "Hello, world!\n")) {
```

Comment:
`responseText &&` is not needed since you're already using `responseText ===`.
And you probably don't even need to check `response.ok`, unless you'd like to treat `Hello, world!\n` responses as invalid for some status codes.

Reply:
Addressed by 019c840